flutter/news/lib/features/frontpage/presentation/bloc/top_headlines_state.dart

```diff
@@ -1,29 +1,14 @@
-part of 'articles_cubit.dart';
-
-abstract class ArticlesState extends Equatable {
-  const ArticlesState();
-}
-
-class TopHeadlinesInitial extends ArticlesState {
-  @override
-  List<Object> get props => [];
-}
-
-class TopHeadlinesLoading extends ArticlesState {
-  @override
-  List<Object> get props => [];
-}
-
-class TopHeadlinesLoaded extends ArticlesState {
-  final List<TopHeadlineViewState> topHeadlines;
-
-  const TopHeadlinesLoaded(this.topHeadlines);
-
-  @override
-  List<Object> get props => [topHeadlines];
-}
-
-class TopHeadlinesError extends ArticlesState {
-  @override
-  List<Object> get props => [];
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:news/features/frontpage/presentation/bloc/top_headlines_viewstate.dart';
+
+part 'top_headlines_state.freezed.dart';
+
+@freezed
+class ArticlesState with _$ArticlesState {
+  const factory ArticlesState.initial() = Initial;
+  const factory ArticlesState.loading() = Loading;
```

Comment:
```suggestion
  const factory ArticlesState.loading() = ArticlesStateLoading;
```

Unfortunately this types will easily conflicts with new ones. 
On another project I'm now using full scoped names to avoid these collisions.

Reply:
Can this be avoided using import alias when using in the same class different state models?

Reply:
Perhaps we don't event need to reference those types externally. In fact I have made them private on 3f2c1773350a8259a757273f12ea0a7b337709c9